Lucid/Stores/RemoteStore.swift

```diff
@@ -81,109 +86,109 @@ public final class RemoteStore<E>: StoringConvertible where E: RemoteEntity {
 
-            case .failure(let error):
-                let error = StoreError.api(error)
-                Logger.log(.error, "\(RemoteStore.self): Error occurred for request path: \(request.config.path): \(error)")
-                completion(.failure(error))
-            }
-        }
+                    let source: RemoteResponseSource = response.cachedResponse ? .urlCache(.empty) : .server(.empty)
 
-        guard hasCachedResponse == false else { return }
+                    if response.isNotModified {
+                        context.set(payloadResult: .success(nil), source: source, for: request.config)
+                        return
+                    }
 
-        let requestCompletion: (APIClientQueueResult<Data, APIError>) -> Void = { result in
-            switch result {
-            case .success(let response):
+                    Task {
+                        do {
+                            try await self.decodingAsyncQueue.enqueue {
+                                do {
+                                    switch context.dataSource {
+                                    case ._remote(.request(_, let endpoint), _, _, _):
+                                        let payload = try E.ResultPayload(from: response.data,
+                                                                          endpoint: endpoint,
+                                                                          decoder: response.jsonCoderConfig.decoder)
+                                        context.set(payloadResult: .success(payload), source: source, for: request.config)
+
+                                    case ._remote(.derivedFromEntityType, _, _, _):
+                                        let payload = try E.ResultPayload(from: response.data,
+                                                                          endpoint: try E.unwrappedEndpoint(for: path),
+                                                                          decoder: response.jsonCoderConfig.decoder)
+                                        context.set(payloadResult: .success(payload), source: source, for: request.config)
+
+                                    case .local,
+                                            .localOr,
+                                            .localThen:
+                                        Logger.log(.error, "\(Self.self): Remote store should not be attempting to handle data source \(context.dataSource).", assert: true)
+                                        return
+                                    }
+
+                                } catch {
+                                    context.set(payloadResult: .failure(.deserialization(error)), source: source, for: request.config)
+                                }
+                            }
+                        } catch {
+                            Logger.log(.error, "\(RemoteStore.self) found error while enqueuing async task")
+                        }
```

Comment:
I think we want to remove the inner `do` on line `112` and fold these two catches into one.

This way we have no scenario where we don't call:

```swift
context.set(payloadResult: .failure(.deserialization(error)), source: source, for: request.config)
```

This is important to make sure any reuse of this context immediately hits this error.